runtime/eventlog.c

```diff
@@ -156,36 +184,63 @@ static void setup_eventlog_file()
                       1, Caml_state->eventlog_out);
     if (ret != 1)
       caml_eventlog_disable();
-    fflush(Caml_state->eventlog_out);
   } else {
     caml_fatal_error("eventlog: could not open trace for writing");
   }
+
+
 }
 #undef OUTPUT_FILE_LEN
 
+static void thread_setup_evbuf()
+{
+  CAMLassert(!evbuf);
+
+  evbuf = caml_stat_alloc_noexc(sizeof(*evbuf));
+
+  if (evbuf == NULL)
+    caml_fatal_error("eventlog: could not allocate event buffer");
```

Comment:
isn't `caml_stat_alloc` and raising Out_of_memory shorter and a more accurate exception here? (the backtrace will reveal where the mem alloc failed)

Reply:
Tracking this for post-merge fixing in https://github.com/ocaml/ocaml/issues/10861